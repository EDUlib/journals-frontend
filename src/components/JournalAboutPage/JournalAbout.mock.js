const mockJournalAbout = {
  accessLength: 0,
  cardImageUrl: '',
  customContent: '<h2>Content Title</h2><p>Custom content details</p>',
  error: null,
  finishedFetching: true,
  heroImageUrl: '',
  journalAboutId: 5,
  journalId: 3,
  longDescription: 'Long description of journal',
  price: '0',
  purchaseUrl: 'http://purchase.example.com/',
  shortDescription: 'Short description of journal',
  startedFetching: false,
  title: 'Title of journal about page',
  structure: [
    {
      id: 6,
      title: 'Title of journal page 1',
      url: 'http://example.com/page1',
      children: [
        {
          id: 8,
          title: 'Title of journal page 1 child',
          url: 'http://example.com/page1child',
          children: [
            {
              id: 9,
              title: 'Title of journal page 1 grand child',
              url: 'http://example.com/page1grandchild',
              children: null,
            },
          ],
        },
      ],
    },
    {
      id: 7,
      title: 'Title of journal page 2',
      url: 'http://example.com/page2',
    },
  ],
};

export default mockJournalAbout;
