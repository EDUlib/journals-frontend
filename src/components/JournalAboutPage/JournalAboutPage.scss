.hero {
  min-height: 280px;
  background-size: cover !important;
  background-repeat: no-repeat !important;
  color: white;
  display: flex;
  flex-direction: column;
  align-items: left;
  justify-content: center;
}

.hero > div {
  margin: 10px 20px;
}

.hero h1 {
  color: white;
}

.hero h2 {
  color: white;
}

.hero .btn{
  font-size: 20px;
  font-weight: bold;
}

.long-description {
  margin-top: 20px;
}
