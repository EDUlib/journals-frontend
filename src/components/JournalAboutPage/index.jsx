import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';


const JournalAboutPage = props => (
  <div className="about-page">
    <h3>{props.title}</h3>
    <Link to={{ pathname: `/${props.journalId}` }}>Enter</Link>
  </div>
);

JournalAboutPage.defaultProps = {
  title: '',
  journalId: 0,
};

JournalAboutPage.propTypes = {
  title: PropTypes.string,
  journalId: PropTypes.number,
};


export default JournalAboutPage;
