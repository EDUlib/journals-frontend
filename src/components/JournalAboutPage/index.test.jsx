import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import JournalAboutPage from './index';
import mockJournalAbout from './JournalAbout.mock';

const JournalAboutPageWrapper = props => (
  <MemoryRouter>
    <JournalAboutPage
      journal={mockJournalAbout}
      serverBaseUrl="example.com/user"
      authorizedJournals={[mockJournalAbout.journalId]}
      {...props}
    />
  </MemoryRouter>
);

describe('<JournalAboutPage />', () => {
  it('renders loading page when page is not finished fetching.', () => {
    const newJournalAbout = Object.assign({}, mockJournalAbout);
    newJournalAbout.startedFetching = true;
    const tree = renderer
      .create((
        <JournalAboutPageWrapper
          journal={newJournalAbout}
        />
      )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders page correctly with authorized journal id', () => {
    const tree = renderer
      .create((
        <JournalAboutPageWrapper />
      )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders page correctly without authorized journal id', () => {
    const tree = renderer
      .create((
        <JournalAboutPageWrapper
          authorizedJournals={[]}
        />
      )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the empty page.', () => {
    const newJournalAbout = Object.assign({}, mockJournalAbout);
    newJournalAbout.finishedFetching = false;
    const tree = renderer
      .create((
        <JournalAboutPageWrapper
          journal={newJournalAbout}
        />
      )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

