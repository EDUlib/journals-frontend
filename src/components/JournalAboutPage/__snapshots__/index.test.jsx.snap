// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JournalAboutPage /> renders loading page when page is not finished fetching. 1`] = `"Loading..."`;

exports[`<JournalAboutPage /> renders page correctly with authorized journal id 1`] = `
<div>
  <div
    className="hero"
    style={
      Object {
        "background": "linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5) ), url()",
      }
    }
  >
    <h1>
      Title of journal about page
    </h1>
    <h2>
      Short description of journal
    </h2>
    <div>
      <a
        className="btn btn-success"
        href="/5"
        onClick={[Function]}
      >
        View Now
      </a>
    </div>
  </div>
  <div>
    <p
      className="long-description"
    >
      Long description of journal
    </p>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h2>Content Title</h2><p>Custom content details</p>",
        }
      }
    />
  </div>
</div>
`;

exports[`<JournalAboutPage /> renders page correctly without authorized journal id 1`] = `
<div>
  <div
    className="hero"
    style={
      Object {
        "background": "linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5) ), url()",
      }
    }
  >
    <h1>
      Title of journal about page
    </h1>
    <h2>
      Short description of journal
    </h2>
    <div>
      <a
        className="btn btn-success"
        href="example.com/userhttp://purchase.example.com/"
        onClick={[Function]}
        target="_self"
      >
        Purchase Access ($0) for unlimited time
      </a>
    </div>
  </div>
  <div>
    <p
      className="long-description"
    >
      Long description of journal
    </p>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h2>Content Title</h2><p>Custom content details</p>",
        }
      }
    />
  </div>
</div>
`;

exports[`<JournalAboutPage /> renders the empty page. 1`] = `null`;
